test/test_s2i.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/python3
+import logging
+import os
+
+from conu import DockerBackend, S2IDockerImage
+
+import pytest
+
+image_name = os.environ.get("IMAGE_NAME", "ruby")
+test_dir = os.path.abspath(os.path.dirname(__file__))
+puma_app_path = os.path.join(test_dir, "puma-test-app")
+rack_app_path = os.path.join(test_dir, "rack-test-app")
+
+
+backend = DockerBackend(logging_level=logging.DEBUG)
+
+
+@pytest.mark.parametrize("app_path", [
+    puma_app_path,
+    rack_app_path,
+])
+def test_s2i_extending(app_path):
+    i = S2IDockerImage(image_name)
+    app_name = os.path.basename(app_path)
+    app = i.extend(app_path, app_name)
+    try:
+        c = app.run_via_binary()
+        try:
+            c.wait_for_port(8080)
+            assert c.is_port_open(8080)
+            response = c.http_request("/", port="8080")
+            assert response.ok
+        finally:
+            c.stop()
+            c.wait()
+            # debugging
+            print(c.logs())
+            c.delete()
+    finally:
+        app.rmi()
+
+
+def test_usage():
+    i = S2IDockerImage(image_name)
+    c = i.run_via_binary()
+    c.wait()
+    logs = c.logs().decode("utf-8").strip()
```

Comment:
It seems, on my system, c.logs() doesn't return empty lines ('\n' instead of '\n\n'). Are you also experiencing this? If so, is this expected?

Reply:
I think I'm getting the same result:

```
>       assert logs == usage
E       AssertionError: assert 'This is a S2...by-sample-app' == 'This is a S2I...by-sample-app'
E           This is a S2I ruby-2.4 Fedora base image:
E           To use it, install S2I: https://github.com/openshift/source-to-image
E         +
E           Sample invocation:
E         +
E           s2i build https://github.com/container-images/ruby.git --context-dir=test/puma-test-app/ modularitycontainers/ruby-24 ruby-sample-app
E         +
E         +
E           You can then run the resulting image via:
E           docker run -p 8080:8080 ruby-sample-app
```

Reply:
This is actually an issue in docker -- it can't handle logs with subsequent multilines (!)

```
$ docker run --name asd fedora:27 printf "asd\n\n\nqwe\n"
asd


qwe
$ docker logs asd
asd
qwe
```

I'll report this upstream (once I'll be able to reproduce on vanilla docker). Will workaround in the meantime.